mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraUserMailboxRightsDAO.java

```diff
@@ -98,12 +99,15 @@ private PreparedStatement prepareSelectAllForUser(Session session) {
             .where(eq(USER_NAME, bindMarker(USER_NAME))));
     }
 
-    public CompletableFuture<Void> update(CassandraId cassandraId, PositiveUserACLChanged positiveUserAclChanged) {
+    public ACLDiff update(CassandraId cassandraId, PositiveUserACLChanged positiveUserAclChanged) {
         PositiveUserACLDiff positiveUserAclDiff = PositiveUserACLDiff.computeDiff(positiveUserAclChanged.getOldACL(), positiveUserAclChanged.getNewACL());
-        return CompletableFuture.allOf(
+        CompletableFuture.allOf(
             addAll(cassandraId, positiveUserAclDiff.addedEntries()),
             removeAll(cassandraId, positiveUserAclDiff.removedEntries()),
-            addAll(cassandraId, positiveUserAclDiff.changedEntries()));
+            addAll(cassandraId, positiveUserAclDiff.changedEntries()))
+            .join();
+
+        return positiveUserAclDiff.getAclDiff();
```

Comment:
I guess the positiveUserAclDiff should be generated from the ACL diff, and not the other way around.

IE we should IMO remove  positiveUserAclDiff.getAclDiff() method